compiler/crates/relay-codegen/src/build_ast.rs

```diff
@@ -1269,7 +1269,8 @@ impl<'schema, 'builder, 'config> CodegenBuilder<'schema, 'builder, 'config> {
         let DeferDirective { if_arg, label_arg } = DeferDirective::from(
             defer,
             &self.project_config.schema_config.defer_stream_interface,
-        );
+        )
+        .unwrap();
```

Comment:
@captbaritone I was able to capture the Diagnostic errors in the other places `DeferDirective`/`StreamDirective` are used, but I'm not sure how to handle it here. This change doesn't stop the panics because I think this will panic before it gets to the transforms where the errors are handled. 

It doesn't look like this file captures errors and I think it might be a big change to introduce that, but my Rust experience is limited so let me know how you think we should handle this?